### PR TITLE
Improve environment testing and code

### PR DESCRIPTION
**Pull request by rickporter-tuono**
1. CLI now fails when attempting to change parent of existing environment
2. Add test for environment inheritance
3. Add test to verify each environment can have different parameter values

### CODE REVIEW

File: tests/pytest/test_parameters.py [hunk 1]
```diff
@@ -326,6 +326,114 @@ def test_project_separation(self):
...
+
+        result = self.run_cli(cmd_env, env3_list)
+        self.assertIn(f"{var1_name},{var1_value3},{env_name3}", result.out())
+        self.assertIn(f"{var2_name},{var2_value3},{env_name3}", result.out())
+
+        docker_cmd = " param export docker -s"
+        result = self.run_cli(cmd_env, proj_cmd + docker_cmd)
+        self.assertEqual(result.return_value, 0)
+        self.assertEqual(result.out(), f"""\
+{var1_name.upper()}={var1_value1}
```
Comment (jeking3) [inline tests/pytest/test_parameters.py:411]: Order does not matter so one could test to make sure each line is present.

File: src/main.rs [hunk 2]
```diff
@@ -442,18 +442,18 @@ fn process_environment_command(
                     "Environment '{}' not updated: same description",
                     env_name.unwrap()
                 ))?;
+            } else if parent_name.is_some() && parent_name.unwrap() != details.parent.as_str() {
+                error_message(format!(
+                    "Environment '{}' parent cannot be updated.",
+                    env_name.unwrap()
+                ))?;
+                process::exit(6);
```
Comment (jeking3) [inline src/main.rs:450]: Is this like exit(42) (some arbitrary non-zero number)?

**Review by jeking3 — APPROVED** [review]

File: src/main.rs (hunk 2 above)
Comment (rickporter-tuono) [reply src/main.rs:450]: Yes, just trying to give diversity in the value returned to maybe help isolate with less than full debug info.

File: tests/pytest/test_parameters.py (hunk 1 above)
Comment (rickporter-tuono) [reply tests/pytest/test_parameters.py:411]: Let's chat about this... We do order the `param export` output on the server side, although the order will likely change soon. We should understand what cli tables/lists we can depend on the order of and probably fix all of them.

**Pull request merged**